sorts.js

```diff
@@ -3,3 +3,101 @@ var group2 = [32, 29, 28, 47, 50, 43, 40, 45, 37, 46, 27, 49, 35, 26, 34, 44, 48
 var group3 = ['sadly', 'roast', 'Loopy', 'burger', 'gavel', 'lure', 'postpone', 'frontal', 'Gavel', 'lavender'];
 
 // Your code goes here! Hooray!
+
+function bubbleSort(oldArray) {
+   
+   var newArray = oldArray.slice();
+   var arraySize = newArray.length;
+   var swap;
+   
+   // To convert all strings to lowercase
+   for (var j = 0; j < arraySize; j++) {
+      if (typeof newArray[j] === 'string') { 
+      newArray[j] = newArray[j].toLowerCase();
+      }
+   }
+   
+   for (var i = 0; i < (arraySize - 1); i++) {
+       for (var k = 0; k < (arraySize - i - 1); k++) {
+           if (newArray[k] > newArray[k + 1]) {
+               swap = newArray[k];
+               newArray[k] = newArray[k + 1];
+               newArray[k + 1] = swap;
+           }
+       }
+   }
+   console.log(newArray);
+}
+
+function swap(myArray, indexOne, indexTwo) {
+    
+    var holding = myArray[indexOne];
+    myArray[indexOne] = myArray[indexTwo];
+    myArray[indexTwo] = holding;
+    
+}
+
+function partition(myArray, left, right) {
+
+    var pivot   = myArray[Math.floor((right + left) / 2)],
+        x       = left,
+        y       = right;
+
+
+    while (x <= y) {
+
+        while (myArray[x] < pivot) {
+            x++;
+        }
+
+        while (myArray[y] > pivot) {
+            y--;
+        }
+
+        if (x <= y) {
+            swap(myArray, x, y);
+            x++;
+            y--;
+        }
+    }
+    return x;
+}
+
+function quickSort(oldArray, left, right) {
+
+    var newArray = oldArray.slice();
+    var index;
+    
+    // To convert all strings to lowercase
+   for (var j = 0; j < newArray.length; j++) {
+      if (typeof newArray[j] === 'string') { 
+      newArray[j] = newArray[j].toLowerCase();
+      }
+   }
+
+    if (newArray.length > 1) {
+
+        left = typeof left != "number" ? 0 : left;
+        right = typeof right != "number" ? newArray.length - 1 : right;
+
+        index = partition(newArray, left, right);
+
+        if (left < index - 1) {
+            quickSort(newArray, left, index - 1);
+        }
+
+        if (index < right) {
+            quickSort(newArray, index, right);
+        }
+
+    }
+   console.log(newArray);
```

Comment:
It looks like you're doing this recursively, so you'll want to return the contents of your array so you can concatenate all the pieces together. The other thing is that you need and end condition since this is recursive. The end case is an empty array.